### PR TITLE
ci: Initial use of GitHub Actions.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This builds via cmake on Linux, macOS, and Windows. It doesn't yet build the tools that use Qt or run tests.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on: [push, pull_request]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build-vmake:
```
Comment (zayenz) [inline .github/workflows/build.yml]: I'm guessing this is intended to be `build-cmake`?

Comment (waywardmonkeys) [reply .github/workflows/build.yml]: Yes, force pushed with a fix.

**Comment by waywardmonkeys** [comment]
Checking back in on this (and my other PRs) ...

**Comment by zayenz** [comment]
I've not found a way to test this in a meaningful manner without merging, so merging provisionally.

**Comment by zayenz** [comment]
Seems to work from runs in #181 and #180

**Pull request merged**